cli/asc.js

```diff
@@ -270,6 +271,64 @@ exports.main = function main(argv, options, callback) {
 
   // Set up base directory
   const baseDir = args.baseDir ? path.resolve(args.baseDir) : ".";
+  const target = args.target;
+
+  // Once the basedir is calculated, we can resolve the config, and it's extensions
+  let asconfig = getAsconfig(args.config, baseDir, readFile);
+  let asconfigDir = baseDir;
+
+  while (asconfig) {
+    // merge target first, then merge options, then merge extended asconfigs
+    if (asconfig.targets && asconfig.targets[target]) {
+      args = optionsUtil.merge(exports.options, args, asconfig.targets[target]);
+    }
+    if (asconfig.options) {
+      args = optionsUtil.merge(exports.options, args, asconfig.options);
+    }
+
+    // entries are added to the compilation
+    if (asconfig.entries) {
+      for (const entry of asconfig.entries) {
+        argv.push(
+          path.isAbsolute(entry)
+            ? entry
+            // the entry is relative to the asconfig directory
+            : path.join(asconfigDir, entry)
+        );
+      }
+    }
+
+    // asconfig "extends" another config, merging options of it's parent
+    if (asconfig.extends) {
+      asconfigDir = path.isAbsolute(asconfig.extends)
+        // absolute extension path means we know the exact directory and location
+        ? path.dirname(asconfig.extends)
+        // relative means we need to calculate a relative asconfigDir
+        : path.join(asconfigDir, path.dirname(asconfig.extends));
+      asconfig = getAsconfig(path.basename(asconfig.extends), asconfigDir, readFile);
+    } else {
+      asconfig = null; // finished resolving the configuration chain
+    }
+  }
+
+  // Just some helper functions
+  const resolve = arg => path.isAbsolute(arg)
+    ? arg
+    // this transform is relative to process.cwd()
+    : path.resolve(path.join(baseDir, arg));
+  const unique = () => {
+    const set = new Set();
+    return arg => {
+      if (set.has(arg)) return false;
+      set.add(arg);
+      return true;
+    };
+  };
```

Comment:
These seem useful but conflict a little with other code in style. Would it be possible to make these normal functions?

Reply:
Should I put them at the bottom too?

Reply:
I don't particularly mind the location. Perhaps just what fits better with other surrounding code.

Reply:
All set!